wallet/wallet-controller/src/lib.rs

```diff
@@ -123,6 +123,18 @@ impl<T: NodeInterface + Clone + Send + Sync + 'static> Controller<T> {
         Ok(wallet)
     }
 
+    pub fn encrypt_wallet(&mut self, password: &Option<String>) -> Result<(), ControllerError<T>> {
+        self.wallet.encrypt_wallet(password).map_err(ControllerError::WalletError)
+    }
+
+    pub fn unlock_wallet(&mut self, password: &Option<String>) -> Result<(), ControllerError<T>> {
```

Comment:
Same here... unlock without a password is kind of weird and needs explanation.

Reply:
removed Optional